scripts/gas_stats.py

```diff
@@ -744,6 +740,185 @@ def borrowActions(env, withdrawCash, redeem):
+        ],
+        depositActionAmount=2e18,
+    )
+    env.notional.batchBalanceAndTradeAction(
+        accounts[1], [collateral, borrowAction], {"from": accounts[1], "value": 2e18})
+
+    # Drop ETH price
+    answer = env.ethOracle["DAI"].latestAnswer()
+    env.ethOracle["DAI"].setAnswer(0.017e18)
+
+    liquidatedPortfolioBefore = env.notional.getAccountPortfolio(accounts[1])
+    maturities = [asset[1] for asset in liquidatedPortfolioBefore]
+    txn = env.notional.liquidatefCashCrossCurrency(accounts[1], 2, 1, maturities, [0, 0], {"from": accounts[2]})
+    log_gas("fcashCrossCurrencyLiquidate.fCashAssets", txn, txn)
+
+    # Restore previous price
+    env.ethOracle["DAI"].setAnswer(answer)
+
+def cashLiquidateSetup(env):
+    env.notional.batchBalanceAction(
+        accounts[0],
+        [
+            get_balance_action(2, "DepositAssetAndMintNToken", depositActionAmount=5000000e8),
+            get_balance_action(4, "DepositAssetAndMintNToken", depositActionAmount=5000000e8),
+        ],
+        {"from": accounts[0]},
+    )
+    env.notional.initializeMarkets(2, True)
+    env.notional.initializeMarkets(4, True)
+
+def localLiquidate(env):
+    cashLiquidateSetup(env)
+    batch = get_balance_trade_action(
+        2,
+        "DepositAssetAndMintNToken",
+        [{"tradeActionType": "Borrow", "marketIndex": 1, "notional": 8e8, "maxSlippage": 0}],
+        withdrawEntireCashBalance=True,
+        depositActionAmount=500e8,
+        redeemToUnderlying=True,
+    )
+    env.notional.batchBalanceAndTradeAction(
+        accounts[1], [batch], {"from": accounts[1]}
+    )
+    tokenDefaults = nTokenDefaults["Collateral"]
+    tokenDefaults[1] = 70
+    env.notional.updateTokenCollateralParameters(2, *(tokenDefaults))
+    txn = env.notional.liquidateLocalCurrency(accounts[1].address, 2, 0, {"from": accounts[2]})
+    log_gas("liquidateLocalCurrency.transferCash", txn, txn)
+
+def collateralLiquidate(env):
+    cashLiquidateSetup(env)
+    borrowAction = get_balance_trade_action(
+        2,
+        "None",
+        [{"tradeActionType": "Borrow", "marketIndex": 1, "notional": 100e8, "maxSlippage": 0}],
+        withdrawEntireCashBalance=True,
+        redeemToUnderlying=True,
+    )
+    ethCollateral = get_balance_trade_action(1, "DepositUnderlying", [], depositActionAmount=3e18)
+    env.notional.batchBalanceAndTradeAction(
+        accounts[1], [ethCollateral, borrowAction], {"from": accounts[1], "value": 3e18}
+    )
+
+    # Drop ETH price
+    answer = env.ethOracle["DAI"].latestAnswer()
+    env.ethOracle["DAI"].setAnswer(0.025e18)
+    txn = env.notional.liquidateCollateralCurrency(accounts[1], 2, 1, 0, 0, False, False, {"from": accounts[2]})
+    log_gas("liquidateCollateralCurrency.transferCash", txn, txn)
+    chain.undo(1)
+    txn = env.notional.liquidateCollateralCurrency(accounts[1], 2, 1, 0, 0, True, False, {"from": accounts[2]})
+    log_gas("liquidateCollateralCurrency.withdrawTokens", txn, txn)
```

Comment:
Can you add another version here and rename they keys so we have:
```
liquidateCollateralCurrency.transferCash
liquidateCollateralCurrency.transferAndWithdrawCash
liquidateCollateralCurrency.transferWithdrawRedeemCash
```

And then can we do a separate liquidation tests here the collateral is in nTokens? So the setup will be the same but instead of "DepositUnderlying" you do "DepositUnderlyingAndMintNToken".

We can key that one as: `liquidateCollateralCurrency.transferNToken`

Reply:
done